### PR TITLE
Feature/cdcinitializer 93

### PR DESCRIPTION
**Pull request by bmmsantos**

### CODE REVIEW

File: cdc-accelerator/preview/preview.js [hunk 1]
```diff
@@ -4,266 +4,46 @@
...
-    if (!useLocalWebSdk) {
-        return appendGigyaScriptTag({ apiKey, lang })
-    }
-
-    const httpClient = new XMLHttpRequest()
-    httpClient.open('GET', await getScreenSetWebSdkFilename({ apiKey }))
-    httpClient.onreadystatechange = function () {
-        if (this.readyState == 4 && this.status == 200) {
-            appendGigyaScriptTag({ apiKey, webSdk: httpClient.responseText, lang })
+    Navigation.loadSiteSelector({ apiKey, origin }).then((resp) => {
```
Comment (luismarques-io) [inline cdc-accelerator/preview/preview.js:27]: Remove unused `resp` param.

File: cdc-accelerator/preview/preview.js [hunk 2]
```diff
@@ -328,140 +108,492 @@ const loadScreenSetsMenu = (callback = () => {}) => {
...
+
+        // Remove comments and \n
+        screenJs = screenJs.replace(/\/\*[\s\S]*?\*\/|([^\\:]|^)\/\/.*$/gm, '')
+        screenJs = '' + screenJs.replaceAll('\n', '')
+
+        let screenSetEvents
+        try {
+            screenSetEvents = eval('(' + screenJs + ')')
+        } catch (error) {
+            alert('Error loading local Screen-Set events from file: \n\n' + filename)
```
Comment (luismarques-io) [inline cdc-accelerator/preview/preview.js:481]: Replace with `location.reload()`, to enable the build command to finish.

Also find other locations where we are throwing the error to the console and replace them with a reload, ex: webSdk file, emailTemplates metadata

**Review by luismarques-io — CHANGES_REQUESTED** [review]

File: cdc-accelerator/preview/preview.js [hunk 3]
```diff
@@ -328,140 +108,492 @@ const loadScreenSetsMenu = (callback = () => {}) => {
...
     }
-    filename += `webScreenSets/${screenSetID}/${screenSetID}.js`
 
-    return await getScreenSetEventsFromFile(filename)
-}
+    static #createIFrame() {
+        const iFrameElement = document.createElement('iframe')
+        iFrameElement.id = 'cdc-initializer--preview-container_iframeEmails'
+        iFrameElement.style.width = '800px'
+        iFrameElement.style.height = '800px'
```
[the remark this answers is not in the document]
Comment (bmmsantos) [reply cdc-accelerator/preview/preview.js]: Done

File: src/index.html
[the remark this answers is not in the document]
Comment (bmmsantos) [reply src/index.html:1]: Done

File: cdc-accelerator/preview/preview.js (hunk 1 above)
Comment (bmmsantos) [reply cdc-accelerator/preview/preview.js]: Done

File: cdc-accelerator/preview/preview.js (hunk 2 above)
Comment (bmmsantos) [reply cdc-accelerator/preview/preview.js]: Done

**Review by luismarques-io — APPROVED** [review]

**Pull request merged**